backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -173,30 +224,57 @@ public LiveOktaRepository(
     User user = users.single();
 
     String orgId = org.getExternalId();
-
+    boolean userInOrg = false;
     // Remove user from old groups
-    Set<String> roleGroupsToRemove =
-        Stream.of(OrganizationRole.values())
-            .filter(r -> r != OrganizationRole.getDefault())
-            .map(r -> generateGroupName(orgId, r))
-            .collect(Collectors.toSet());
     for (Group g : user.listGroups()) {
       if (g.getType() == GroupType.OKTA_GROUP
-          && roleGroupsToRemove.contains(g.getProfile().getName())) {
-        g.removeUser(user.getId());
+          && g.getProfile().getName().startsWith(generateGroupOrgPrefix(orgId))) {
+        // do not remove user from org's default group
+        if (g.getProfile()
+            .getName()
+            .equals(generateRoleGroupName(orgId, OrganizationRole.getDefault()))) {
```

Comment:
Nit: this string and the one on L231 are constant over the iteration, should be factored out to avoid repeated string-building.